### PR TITLE
fix: update npx command in setup.js

### LINKED ISSUE #316: 🐛 Bug: npx fails in setup.js
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

`pnpm setup` exits successfully.

### Actual

It fails:

```js
Generating all-contributors table in README.md...
Error: 
    at file:///Users/jp/dev/ens-expire/script/setup.js:184:9
    exit code: 1
    at ChildProcess.<anonymous> (file:///Users/jp/.npm/_npx/721a1717bb84573e/node_modules/zx/build/core.js:146:26)
    at ChildProcess.emit (node:events:513:28)
    at ChildProcess.emit (node:domain:489:12)
    at maybeClose (node:internal/child_process:1091:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:513:28)
    at Socket.emit (node:domain:489:12)
    at Pipe.<anonymous> (node:net:322:12)
    at Pipe.callbackTrampoline (node:internal/async_hooks:130:17)

Removing devDependency packages only used for setup...
✔️ Done.

Looks like there was a problem. Correct it and try again? 😕
```

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by jsjoeio**
## PR Checklist

- [x] Addresses an existing open issue: fixes #316
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

If the `all-contributors` package is not already installed on the local machine, the script exits. We can pass `--yes` to autoinstall it.

### CODE REVIEW

File: script/setup.js [hunk 1]
```diff
@@ -181,7 +181,7 @@ try {
 	console.log();
 	console.log(chalk.gray`Generating all-contributors table in README.md...`);
 
-	await $`npx all-contributors generate`;
+	await $`npx --yes all-contributors generate`;
```
Comment (JoshuaKGoldberg) [inline script/setup.js:184]: Aha! So this was the culprit! 

Looking at the other used in this script, they're all added as `devDependency` entries in `package.json`. `all-contributors` is the only one installed with `npx`. What do you think about installing it as a dev dependency @jsjoeio? I'm leaning towards that so the setup script doesn't have to install any additional packages.

If so, note that it'll need to be uninstalled in the `pnpm uninstall` command later on in the file, as it wouldn't be used by users of the package.

If not, this is good to merge IMO!

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thank you so much for pinpointing this! Just a hunch from me on an alternate solution, otherwise this is great! Thanks @jsjoeio!

File: script/setup.js (hunk 1 above)
Comment (jsjoeio) [reply script/setup.js:184]: That sounds good to me!

File: script/setup.js [hunk 2]
```diff
@@ -302,7 +302,7 @@ try {
 	console.log(
 		chalk.gray`Removing devDependency packages only used for setup...`
 	);
-	await $`pnpm remove @clack/prompts chalk octokit replace-in-file -D`;
+	await $`pnpm remove @clack/prompts chalk octokit replace-in-file all-contributors-cli -D`;
```
Comment (JoshuaKGoldberg) [inline script/setup.js:305]: Don't mind me 😄 just being nitpicky...

```suggestion
	await $`pnpm remove @clack/prompts all-contributors-cli chalk octokit replace-in-file -D`;
```

**Review by JoshuaKGoldberg — APPROVED** [review]
🙌 this is great, thanks @jsjoeio!

File: package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
 		"@typescript-eslint/eslint-plugin": "^5.48.2",
 		"@typescript-eslint/parser": "^5.48.2",
 		"@vitest/coverage-istanbul": "^0.29.0",
+		"all-contributors-cli": "^6.24.0",
```
Comment (JoshuaKGoldberg) [inline package.json:44]: Ahh, one more thing here, [Knip]() doesn't know that this is used:

```plaintext
Run pnpm lint:knip

> template-typescript-node-package@1.2[7](https://github.com/JoshuaKGoldberg/template-typescript-node-package/actions/runs/4484576327/jobs/7885225024?pr=315#step:4:8).5[8](https://github.com/JoshuaKGoldberg/template-typescript-node-package/actions/runs/4484576327/jobs/7885225024?pr=315#step:4:9) lint:knip /home/runner/work/template-typescript-node-package/template-typescript-node-package
> knip

Unused dev dependencies (1)
all-contributors-cli  package.json
 ELIFECYCLE  Command failed with exit code 1.
```

The root issue is that Knip doesn't recognize [zx](https://github.com/google/zx) scripts. I'll try to see if I can contribute that!

Comment (JoshuaKGoldberg) [reply package.json:44]: https://github.com/webpro/knip/issues/96

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @jsjoeio for tool

**Pull request merged**